src/Presentation/Memo4me.UI/Extensions/AddAuthOpenIdExtension.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Memo4me.UI.Extensions;
+
+public static class AddAuthOpenIdExtension
+{
+    public static void AddAuthOpenId(this IServiceCollection services,  ConfigurationManager configuration)
+    {
+        services.AddAuthentication(options =>
+            {
+                options.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
+            })
+            .AddCookie()
+            .AddOpenIdConnect("oidc", options =>
+            {
+                options.Authority = configuration["Authentication:OpenIdConnect:Authority"];
+                options.ClientId = configuration["Authentication:OpenIdConnect:ClientId"];
+                options.ClientSecret = configuration["Authentication:OpenIdConnect:ClientSecret"];
+                options.ResponseType = configuration["Authentication:OpenIdConnect:ResponseType"];
+                options.SaveTokens = true;
+                options.RequireHttpsMetadata = false;
```

Comment:
should be configurable, and set to true by default, as it critical for production

Reply:
done